src/api/ledger/parse/orderbook-order.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const _ = require('lodash');
+const utils = require('./utils');
+const flags = utils.core.Remote.flags.offer;
+const parseAmount = require('./amount');
+
+function parseOrderbookOrder(order: Object): Object {
+  const direction = (order.Flags & flags.Sell) === 0 ? 'buy' : 'sell';
+  const takerGetsAmount = parseAmount(order.TakerGets);
+  const takerPaysAmount = parseAmount(order.TakerPays);
+  const quantity = (direction === 'buy') ? takerPaysAmount : takerGetsAmount;
+  const totalPrice = (direction === 'buy') ? takerGetsAmount : takerPaysAmount;
+
+  const specification = utils.removeUndefined({
+    direction: direction,
+    quantity: quantity,
+    totalPrice: totalPrice,
+    passive: ((order.Flags & flags.Passive) !== 0) || undefined
+  });
+  // "quality" is omitted intentionally as it corresponds to
```

Comment:
I'm going to be giving a talk on the API once the I finish the first pass of the conversion so that everyone can critique. That will probably be followed by another architects mailing list thread. This API is based on the ripple-http-api v2 proposal which has already been posted to architects, though I don't think we went into this level of detail in that discussion.